src/MongoDB/Manager.c

```diff
@@ -200,6 +200,15 @@ static void php_phongo_manager_prep_uri_options(zval *options TSRMLS_DC) /* {{{
 				continue;
 			}
 
+            if (!strcasecmp(ZSTR_VAL(string_key), MONGOC_URI_READPREFERENCE)) {
+				ZVAL_DEREF(option);
+				SEPARATE_ZVAL_NOREF(option);
+				if (Z_TYPE_P(option) != IS_STRING) {
+                    phongo_throw_exception(PHONGO_ERROR_INVALID_ARGUMENT TSRMLS_CC, "Unsupported readPreference type, a string must be given");
```

Comment:
You make a good point by linking to the [mongodb-user thread](https://groups.google.com/forum/#!msg/mongodb-user/Cun9RklOsR0/DDDtGrh3DwAJ). There are two facets to this, so I'll explain each separately.

PHP serialization only occurs because we're creating a hash key to persist the libmongoc client for the given `MongoDB\Driver\Manager` arguments. The "cannot be serialized" exception therefore has more to do with ReadPreference not supporting `serialize()` than being an invalid argument. If we skipped that hash key creation, a ReadPreference object would [serialize to BSON](https://github.com/mongodb/mongo-php-driver/blob/dd43b08320b35fa3fa3326a8093495dac066ffd2/src/MongoDB/ReadPreference.c#L201) and option validation could report that a document was provided where a string was expected (for `readPreference` URI option). 

My hesitation with validating the options array (i.e. Manager's second argument) is that objects may implement interfaces that affect how they serialize to BSON. At present, `authMechanismProperties` and `readPreferenceTags` are the only non-scalar URI options, but imagine an object implemented `MongoDB\BSON\Serializable` and returned a document or array for those options, respectively. Since BSON is that format we encounter before assigning the options to `mongoc_uri_t`, I think we definitely need validation there and suggest starting with that.

There is no reason for ReadPreference, WriteConcern, and ReadConcern to not support PHP serialization (see: [PHPC-850](https://jira.mongodb.org/browse/PHPC-850)). Once that is implemented, the "cannot be serialized" exception should no longer be an issue (for the hash key generation) and any BSON-level type validation implemented for PHPC-887 will fully cover us.